contrib/database/sql/internal/dsn.go

```diff
@@ -15,6 +15,11 @@ import (
 // ParseDSN parses various supported DSN types into a map of key/value pairs which can be used as valid tags.
 func ParseDSN(driverName, dsn string) (meta map[string]string, err error) {
 	meta = make(map[string]string)
+	if driverName != "mysql" && driverName != "postgres" && driverName != "sqlserver" && strings.Contains(dsn, ":") {
```

Comment:
I think this is a nice change 👍 

Currently the `pgx` one case is missing, but I think we can refactor this we don't make any of these mistakes in the future.

Adding the following in line 40 for the `switch` `default` statement:


```go
// try to parse the DSN and see if the scheme contains a known driver name.
u, err := url.Parse(dsn)
if err != nil {
   // dsn is not a valid URL, so just ignore
  break
}

if driverName != u.Scheme {
   return ParseDSN(u.Scheme, dsn)
}
``` 

Also lets add a few tests covering this new logic into `dsn_test.go`